lighthouse-core/gather/gatherers/seo/font-size.js

```diff
@@ -294,11 +295,19 @@ class FontSize extends Gatherer {
       .sort((a, b) => b.textLength - a.textLength)
       .slice(0, MAX_NODES_SOURCE_RULE_FETCHED)
       .map(async failingNode => {
-        failingNode.cssRule = await fetchSourceRule(driver, failingNode.node);
+        try {
+          const cssRule = await fetchSourceRule(driver, failingNode.node);
+          failingNode.cssRule = cssRule;
+        } catch (err) {
+          // The node was deleted. Don't set `cssRule` in that case.
+          log.error('font-size', err.message);
+        }
         return failingNode;
       });
 
-    const analyzedFailingNodesData = await Promise.all(analysisPromises);
+    const analyzedFailingNodesData = (await Promise.all(analysisPromises))
+      // Throw out the nodes that got deleted.
+      .filter(data => 'cssRule' in data);
```

Comment:
any reason not to do `.filter(data => data.cssRule);`? Usually we reserve `in` for making tsc happy on poorly discriminated unions

Reply:
this is also filtered down on `https://github.com/GoogleChrome/lighthouse/pull/9928/files#diff-aed43c42d5d8fb3c5d70f6775865c8deR351`, so could remove one of them

Reply:
hmm, i think the former check (the one here) needs to keep `undefined` b/c it still counts towards the total text length